src/main/webapp/src/app/map/map.component.ts

```diff
@@ -63,33 +63,99 @@ export class MapComponent {
-      
+      latLng = new google.maps.LatLng(currentHiddenGem.geometry.location.lat,
+        currentHiddenGem.geometry.location.lng);
+
       // Creating a marker and putting it on the map.
       let marker = new google.maps.Marker({
         position: latLng,
         map: this.map,
         title: currentHiddenGem.name,
         icon: diamondIcon
       });
-      
-      let contentString = currentHiddenGem.name + " " + currentHiddenGem.vicinity +
-          ", Star rating: " + currentHiddenGem.rating;
+
+      // Get the photo URL.
+      if (currentHiddenGem.photos != null) {
+        photoUrl = "https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference="
+          + currentHiddenGem.photos[0].photoReference
+          + "&key=AIzaSyCBb8QQBQal9jDNl3ZG6f3bS6ROX2MtYIM";
+      }
+      else {
+        photoUrl = "https://www.flaticon.com/svg/vstatic/svg/3716/3716538.svg?token=exp=1612753174~hmac=f38de8989e705031bdecb2da1464c379"
+      }
+
+      // Get the business type.
+      if (currentHiddenGem.types[0] != null && (currentHiddenGem.types[0] === 'restaurant' || currentHiddenGem.types[0] === 'cafe'))
+        type = currentHiddenGem.types[0];
+      else if (currentHiddenGem.types[1] != null && (currentHiddenGem.types[1] === 'restaurant' || currentHiddenGem.types[1] === 'cafe'))
+        type = currentHiddenGem.types[1];
+
+      // Get the business' opening status.
+      if (currentHiddenGem.openingHours)
+        openingStatus = "Open Now";
+      else
+        openingStatus = "Currently Closed"
+
+      let infoWindowContent = this.createInfoWindow(currentHiddenGem.name, currentHiddenGem.vicinity, photoUrl, type, openingStatus, currentHiddenGem.rating);
 
       let infoWindow = new google.maps.InfoWindow();
-      
+
       marker.addListener("click", () => {
         this.map.setZoom(ON_MARKER_MAP_ZOOM);
         this.map.setCenter(latLng);
-        infoWindow.setContent(contentString);
+        infoWindow.setContent(infoWindowContent);
         infoWindow.setPosition(latLng);
         infoWindow.open(this.map, marker);
       });
     }
   }
+
+  createInfoWindow(name: string, address: string, url: string, businessType: string, openingStatus: string, rating: number) {
+    const divElement = document.createElement('div');
+    const nameElement = document.createElement('h2');
+    nameElement.innerText = name;
+    nameElement.className = "mat-title";
+
+    const addressElement = document.createElement('p');
+    addressElement.innerText = address;
+    addressElement.className = "mat-subheading-1";
+    addressElement.setAttribute('style', 'margin-top: -20px;')
+
+    const imageElement = document.createElement('img');
+    imageElement.src = url;
+    imageElement.alt = name;
+    imageElement.setAttribute('style', 'width:100%; height:100px; object-fit: cover;')
+
+    const detailsDivElement = document.createElement('div');
+    detailsDivElement.setAttribute('style', 'margin-left: 15%')
```

Comment:
Might be too much work for now, but in general would be nice to define the styles for css classes in the stylesheet rather than assigning them inline.

Reply:
I was thinking about doing it, but I wasn't sure how to do it since these are within an info window, and it doesn't work if I add the styling to map.component.css for example

Reply:
O rly? Hmm, it's probably because those styles only apply to the component as per https://angular.io/guide/component-styles and if you create the info window yourself it won't be part of that component.

You could try adding it to the global 'styles.css' file (https://angular.io/guide/component-styles#external-and-global-style-files), though global styles are also not super great. There's probably a way to get this to play nice, but unless @lorikeet-s has suggestions I reckon just stick in a comment to the effect of 'figure out how to put these in a stylesheet' for now.